### PR TITLE
Handle simple Constant expressions in SymbolEnter

### LINKED ISSUE #32689: Enable int-singleton.bal in `jballerina-semtype-test`
**Description:**

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by ushirask**
Partially fixes #32689 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-semtype-test/src/test/java/io/ballerina/types/SemTypeTest.java [hunk 1]
```diff
@@ -53,7 +53,8 @@
                 .filter(name -> name.endsWith(".bal"))
                 .collect(Collectors.toList());
 
-        ignore(testFiles, "int-singleton.bal");
+        //ignore(testFiles, "int-singleton.bal");
```
Comment (rdhananjaya) [inline tests/jballerina-semtype-test/src/test/java/io/ballerina/types/SemTypeTest.java:56]: Shall we delete this 2 lines

Comment (ushirask) [reply tests/jballerina-semtype-test/src/test/java/io/ballerina/types/SemTypeTest.java]: fixed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java [hunk 2]
```diff
@@ -418,7 +429,8 @@ private BLangConstantValue calculateBooleanComplement(BLangConstantValue value)
     }
 
     private BLangConstantValue visitExpr(BLangExpression node) {
-        if (!node.typeChecked) {
+
+        if (!node.typeChecked && !this.semtypeEnabled) {
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java:433]: Please create an issue to check on this before merging this branch to `master`. We need to make sure we don't break the compiler when `semtypeEnabled` is true and we perform the normal compilations.

Comment (ushirask) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java:433]: created in #32778

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 3]
```diff
@@ -496,10 +496,36 @@
...
+            semtype = evaluateConst(constant);
+        }
         semtypeEnv.addTypeDef(constant.name.value, semtype);
     }
 
+    private SemType evaluateConst(BLangConstant constant) {
+        SemType s;
+        switch (constant.symbol.value.type.getKind()) {
+            case INT:
+                s = SemTypes.intConst((long) constant.symbol.value.value);
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java:512]: We usually don't try to force a single return, shall we use `return` just for the sake of being consistent with rest of the file.

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**